recipes/autoconf/all/conanfile.py

```diff
@@ -1,122 +1,146 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-import contextlib
-import os
+from conan import ConanFile
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, rmdir, apply_conandata_patches, replace_in_file
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import unix_path, is_msvc
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.51.3"
 
 
 class AutoconfConan(ConanFile):
     name = "autoconf"
+    description = "Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages"
+    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.gnu.org/software/autoconf/"
-    description = "Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages"
     topics = ("autoconf", "configure", "build")
-    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     settings = "os", "arch", "compiler", "build_type"
 
-    exports_sources = "patches/*"
-
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
-
     @property
     def _settings_build(self):
+        # TODO: Remove for Conan v2
         return getattr(self, "settings_build", self.settings)
 
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+
+    def configure(self):
+        self.win_bash = self._settings_build.os == "Windows"
+
+    def layout(self):
+        basic_layout(self, src_folder="autoconf")
+
     def requirements(self):
         self.requires("m4/1.4.19")
 
-    def build_requirements(self):
-        if hasattr(self, "settings_build"):
-            self.build_requires("m4/1.4.19")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
-
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
+
+    def build_requirements(self):
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+            self.tool_requires("msys2/cci.latest")
+        self.tool_requires("m4/1.4.19")
```

Comment:
it doesn't work with `if hasattr(self, "settings_build"):`? It's important to avoid issues with 1 profile.

Reply:
@SpaceIm I'm not sure what you mean with this comment; `self._settings_build` returns either the `settings_build` if available but falls back on `settings`. So I believe it should work for 1 profile and a build/host profile.

Reply:
1 profile is deprecated, this will help alter consumers to upgrade their usage

Reply:
I disagree with this strategy. It's not a good user experience to break such things brutaly.

Reply:
I'm still using `-pr` only, as I have problems with some packages and `-pr:b and -pr:h`,
so if one package won't with with -pr, and others won't work with -pr:b and -pr:h then we have a gridlock situation.

I'd instead suggest a warning in the conan client with a link to an issue where people can report problematic recipes that they discover using the dual-profiles...

I can start testing again with the dual profiles and look for errors.